src/main.js

```diff
@@ -330,37 +330,49 @@ function initLocales() {
         },
     });
 
-    let defaultLang = state.setting('language');
-    let preferredLangs = _.clone(window.navigator && window.navigator.languages) || [];
+    const setDefaultLanguage = () => {
+        let defaultLang = state.setting('language');
+        let preferredLangs = _.clone(window.navigator && window.navigator.languages) || [];
 
-    // our configs default lang overrides all others
-    if (defaultLang) {
-        preferredLangs.unshift(defaultLang);
-    }
+        // our configs default lang overrides all others
+        if (defaultLang) {
+            preferredLangs.unshift(defaultLang);
+        }
+
+        // set a default language
+        i18next.changeLanguage('en-us');
 
-    // set a default language
-    i18next.changeLanguage('en-us');
+        // Go through our browser languages until we find one that we support
+        for (let idx = 0; idx < preferredLangs.length; idx++) {
+            let lang = preferredLangs[idx];
 
-    // Go through our browser languages until we find one that we support
-    for (let idx = 0; idx < preferredLangs.length; idx++) {
-        let lang = preferredLangs[idx];
+            // if this is a language such as 'fr', add a following one of 'fr-fr' to cover
+            // both cases
+            if (lang.length === 2) {
+                preferredLangs.splice(idx + 1, 0, lang + '-' + lang);
+            }
 
-        // if this is a language such as 'fr', add a following one of 'fr-fr' to cover
-        // both cases
-        if (lang.length === 2) {
-            preferredLangs.splice(idx + 1, 0, lang + '-' + lang);
+            if (_.includes(AvailableLocales.locales, lang.toLowerCase())) {
+                i18next.changeLanguage(lang, (err, t) => {
+                    if (err) {
+                        // setting the language failed so set default again
+                        i18next.changeLanguage('en-us');
+                    }
+                });
+                break;
+            }
         }
+    };
+    setDefaultLanguage();
 
-        if (_.includes(AvailableLocales.locales, lang.toLowerCase())) {
-            i18next.changeLanguage(lang, (err, t) => {
-                if (err) {
-                    // setting the language failed so set default again
-                    i18next.changeLanguage('en-us');
-                }
-            });
-            break;
+    // Update the language if the setting changes.
+    state.$watch('user_settings.language', (lang) => {
+        if (!lang && !state.setting('settings.language')) {
```

Comment:
`state.setting('settings.language')` should be `state.setting('language')` as `.setting()` looks both in `user_settings` and then `settings` already